pkg/volume/event.go

```diff
@@ -77,15 +78,37 @@ func (handler *volumeEventHandler) ObjectType() runtime.Object {
 	return &types.Volume{}
 }
 
-func (handler *volumeEventHandler) Handle(ctx context.Context, _ controller.EventType, object runtime.Object) error {
+func (handler *volumeEventHandler) Handle(ctx context.Context, eventType controller.EventType, object runtime.Object) error {
 	volume := object.(*types.Volume)
 	if !volume.GetDeletionTimestamp().IsZero() {
 		return handler.delete(ctx, volume)
 	}
 
+	if eventType == controller.AddEvent {
```

Comment:
For existing volumes, `Add` event occurs only at start of the controller. This is same as added into `Sync()`. As volumes live for very long, we could add drive name update every day once in future.

Reply:
Do you want to keep this in Update event handler? The reason i added this in Add event handler is because, this can happen only during the node restarts mostly. 

Adding this in Update handler, makes these repeated GET calls to the respective drive objects.

If the drive order can change at runtime (considering it as a rare event), can we suggest them to restart the pods instead of checking everytimea at the update handler?

Reply:
Yep. We will add it later.